src/core/qgsmaplayer.cpp

```diff
@@ -94,6 +94,43 @@ QgsMapLayer::~QgsMapLayer()
   delete mStyleManager;
 }
 
+void QgsMapLayer::clone( QgsMapLayer *layer, bool deep ) const
+{
+  layer->setBlendMode( blendMode() );
+
+  QMap<QString, QgsMapLayerStyle> styles = styleManager()->mapLayerStyles();
```

Comment:
if using range based for loops, you need to have a const copy of the list. Is that the case here?